sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -366,14 +375,138 @@ CosmosAsyncContainer setId(String id) {
     }
 
     /**
-     * Gets a CosmosAsyncItem object without making a service call
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
+     *
+     * @param itemId the item id
+     * @param partitionKey the partition key
+     * @return an {@link Mono} containing the cosmos item response with the read item or an error
+     */
+    public <T> Mono<CosmosAsyncItemResponse<T>>  readItem(String itemId, PartitionKey partitionKey, Class<T> itemType) {
+        return readItem(itemId, partitionKey, new CosmosItemRequestOptions(partitionKey), itemType);
+    }
+
+    /**
+     * Reads an item.
+     * <p>
+     * After subscription the operation will be performed.
+     * The {@link Mono} upon successful completion will contain a cosmos item response with the read item
+     * In case of failure the {@link Mono} will error.
```

Comment:
I have got this feedback from various customers, that on `readItem() API`, in v2 SDK if the item is not found, we used to return `empty Observable()`. 
Because it becomes really difficult for the end user to catch 404 and do nothing. 

@moderakh @kirankumarkolli @mbhaskar what should we do with v4 ?

@mbhaskar do you know what .net behavior is ?
As I have also received feedback that we should be consistent across our SDKs.

Reply:
a response has more than just user data, it has activityId, request-charge, etc.

Although returning Mono.empty() may look nice for the end-user who is only interested in user-data, but doesn't provide the the above response header for more advanced user who is interested in response headers.

Reply:
Thanks for the response @moderakh. @kushagraThapar  make a note of this point, and we can discuss further if required and create a workitem for this.